lang/CMakeLists.txt

```diff
@@ -244,6 +248,23 @@ endif()
 target_link_libraries(libsclang boost_thread_lib boost_system_lib boost_regex_lib boost_filesystem_lib)
 target_include_directories(libsclang PUBLIC ${boost_include_dirs})
 
+
+#target_include_directories(libsclang PUBLIC ${LIBGIT2_INCLUDES} ${LIBGIT2_DEPENDENCY_INCLUDES})
+#target_include_directories(libsclang SYSTEM PRIVATE ${LIBGIT2_SYSTEM_INCLUDES})
+
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(LIBGIT2 REQUIRED IMPORTED_TARGET libgit2)
```

Comment:
I don't think we should be using PkgConfig when building the library ourselves. This might be the reasons of your issues, but I haven't looked closely.

We already have a similar situation in SC codebase - with `portaudio` and `yamlcpp` libraries. They can be built from source or linked from system. You may want to check how portaudio is linked when `SYSTEM_PORTAUDIO` option is off. This configuration is used on the Windows builds by default.